src/main/kotlin/com/exactpro/th2/conn/dirty/fix/MessageLoader.kt

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.conn.dirty.fix
+
+import com.exactpro.th2.SequenceHolder
+import com.exactpro.th2.common.grpc.Direction
+import com.exactpro.th2.common.message.toTimestamp
+import com.exactpro.th2.common.util.toInstant
+import com.exactpro.th2.constants.Constants.IS_POSS_DUP
+import com.exactpro.th2.constants.Constants.MSG_SEQ_NUM_TAG
+import com.exactpro.th2.constants.Constants.POSS_DUP
+import com.exactpro.th2.constants.Constants.POSS_DUP_TAG
+import com.exactpro.th2.dataprovider.grpc.DataProviderService
+import com.exactpro.th2.dataprovider.grpc.MessageGroupResponse
+import com.exactpro.th2.dataprovider.grpc.MessageSearchRequest
+import com.exactpro.th2.dataprovider.grpc.MessageSearchResponse
+import com.exactpro.th2.dataprovider.grpc.MessageStream
+import com.exactpro.th2.dataprovider.grpc.TimeRelation
+import com.exactpro.th2.lme.oe.util.ProviderCall
+import com.google.protobuf.Int32Value
+import com.google.protobuf.Timestamp
+import com.google.protobuf.util.Timestamps.compare
+import io.netty.buffer.ByteBuf
+import io.netty.buffer.Unpooled
+import java.time.Instant
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.OffsetTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import kotlin.math.ceil
+import mu.KotlinLogging
+
+class MessageLoader(
+    private val dataProvider: DataProviderService,
+    private val sessionStartTime: LocalTime?
+) {
+    private var sessionStart = OffsetDateTime
+        .now(ZoneOffset.UTC)
+        .with(sessionStartTime ?: OffsetTime.now(ZoneOffset.UTC))
+    private var sessionStartTimestamp = sessionStart
+        .toInstant()
+        .toTimestamp()
+    private var previousDaySessionStart = sessionStart
+        .minusDays(1)
+        .toInstant()
+        .toTimestamp()
+
+    fun updateTime() {
+        sessionStart = OffsetDateTime
+            .now(ZoneOffset.UTC)
+            .with(OffsetTime.now(ZoneOffset.UTC))
+        sessionStartTimestamp = Instant.now().toTimestamp()
+        previousDaySessionStart = Instant.now().toTimestamp()
+    }
```

Comment:
This method is used nowhere, let's remove it and use val instead of var for sessionStart, sessionStartTimestamp, previousDaySessionStart declaration